src-tauri/src/main.rs

```diff
@@ -0,0 +1,150 @@
+// Prevents additional console window on Windows in release, DO NOT REMOVE!!
+#![cfg_attr(not(debug_assertions), windows_subsystem = "windows")]
+
+use std::env;
+use std::io;
+use std::path::Path;
+use std::sync::OnceLock;
+use std::time::Duration;
+
+use crate::depotdownloader::{get_depotdownloader_url, DEPOTDOWNLOADER_VERSION};
+use crate::terminal::Terminal;
+use tauri::{AppHandle, Emitter};
+
+mod depotdownloader;
+mod steam;
+mod terminal;
+
+/// The first terminal found. Used as default terminal.
+static TERMINAL: OnceLock<Vec<Terminal>> = OnceLock::new(); // We create this variable now, and quickly populate it in preload_vectum(). we then later access the data in start_download()
+
+/// This function is called every time the app is reloaded/started. It quickly populates the [`TERMINAL`] variable with a working terminal.
+#[tauri::command]
+async fn preload_vectum(app: AppHandle) {
+    // Only fill this variable once.
+    if TERMINAL.get().is_none() {
+        TERMINAL
+            .set(terminal::get_installed_terminals(true).await)
+            .expect("Failed to set available terminals")
+    }
+
+    // Send the default terminal name to the frontend.
+    app.emit(
+        "default-terminal",
+        Terminal::pretty_name(&TERMINAL.get().unwrap()[0]),
+    )
+    .unwrap();
+}
+
+#[tauri::command]
+async fn start_download(steam_download: steam::SteamDownload) {
+    let default_terminal = TERMINAL.get().unwrap();
+    let working_dir = env::current_dir().unwrap();
+
+    let terminal_to_use = if steam_download.options().terminal().is_none() { default_terminal.first().unwrap() } else { &Terminal::from_index(&steam_download.options().terminal().unwrap()).unwrap() };
+
+    println!("-------------------------DEBUG INFO------------------------");
+    println!("received these values from frontend:");
+    println!("\t- Username: {}", steam_download.username().as_ref().unwrap_or(&String::from("Not provided")));
+    // println!("\t- Password: {}", steam_download.password().as_ref().unwrap_or(&String::from("Not provided"))); Don't log in prod lol
+    println!("\t- App ID: {}", steam_download.app_id());
+    println!("\t- Depot ID: {}", steam_download.depot_id());
+    println!("\t- Manifest ID: {}", steam_download.manifest_id());
+    println!("\t- Output Path: {}", steam_download.output_path());
+    println!("------------------------DEBUG INFORMATION-----------------");
```

Comment:
`DEBUG INFO` and `DEBUG INFORMATION` ?
Would probably remove the 2nd one. Don't see a reason for it and looks little strange.

Reply:
Oops²